core-it-support/core-it-bootstrap/src/main/java/org/apache/maven/its/bootstrap/DownloadMojo.java

```diff
@@ -0,0 +1,153 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
+import org.apache.maven.project.ProjectBuildingRequest;
+import org.apache.maven.shared.transfer.dependencies.resolve.DependencyResolver;
+import org.apache.maven.shared.transfer.dependencies.resolve.DependencyResolverException;
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * Boostrap plugin to download all required dependencies
+ */
+@Mojo( name = "download" )
+public class DownloadMojo
+        extends AbstractMojo
+{
+
+    /**
+     * A list of artifacts coordinates.
+     */
+    @Parameter
+    private List<Dependency> dependencies = new ArrayList<>();
+
+    /**
+     * A list of string of the form groupId:artifactId:version[:packaging[:classifier]].
+     */
+    @Parameter
+    private List<String> artifacts = new ArrayList<>();
+
+    /**
+     * A file containing lines of the form groupId:artifactId:version[:packaging[:classifier]].
+     */
+    @Parameter
+    private File file;
+
+    @Component
+    private DependencyResolver dependencyResolver;
+
+    @Component
+    private MavenSession session;
+
+    @Override
+    public void execute() throws MojoFailureException
+    {
+        if ( file != null && file.exists() )
+        {
+            try ( BufferedReader reader = new BufferedReader( new FileReader( file ) ) )
+            {
+                while ( true )
+                {
+                    String line = reader.readLine();
+                    if ( line != null )
+                    {
+                        artifacts.add( line );
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+            catch ( IOException e )
+            {
+                throw ( MojoFailureException ) new MojoFailureException(
+                                "Unable to read dependencies: " + file ).initCause( e );
+            }
+        }
+        for ( String artifact : artifacts )
+        {
+            if ( artifact != null )
+            {
+                String trimmed = artifact.trim();
+                if ( !trimmed.isEmpty() && !trimmed.startsWith( "#" ) )
```

Comment:
I see here, bu they can be dropped upfront, no?

Reply:
You're right, that does not make much sense.  I've moved it earlier.